src/app/wallets/add-invoice-for-wallet.ts

```diff
@@ -0,0 +1,100 @@
+import { toSats } from "@domain/bitcoin"
+import { invoiceExpirationForCurrency } from "@domain/bitcoin/lightning"
+import { MakeWalletInvoiceFactory } from "@domain/wallet-invoices/wallet-invoice-factory"
+import { MakeLndService } from "@services/lnd"
+import { MakeWalletsRepository, MakeWalletInvoicesRepository } from "@services/mongoose"
+
+export const addInvoiceForSelf = async ({
+  walletId,
+  memo,
+  amount,
+}: AddInvoiceSelfArgs): Promise<LnInvoice | ApplicationError> => {
+  if (!(amount && amount > 0))
+    return new Error("Incorrect method used for no-amount invoice")
```

Comment:
I wasn't sure what type of `Error` I should use here and which class I should extend from ... would be something like a `BadArgumentError`?

`ApplicationError` is already defined as the union of some `DomainError` sub-types, so it doesn't seem like this would derive from that

Reply:
I would put this into a `checkedToSats` method that returns `Satoshi | ConversionError`. This method can be called in the gql layer prior to calling the use case.